### DISCUSSION TITLE
Undefined array key "primaryKey" while running  php artisan site-search:crawl

### DISCUSSION THREAD

**Discussion by Amik-TJ** [spatie/laravel-site-search#23] [Q&A]
C:\xampp\htdocs\Laravel_Test\laravel-sitemap>php artisan site-search:crawl
Dispatching job to crawl `https://hilinkz.com/`

   ErrorException 

  Undefined array key "primaryKey"

  at C:\xampp\htdocs\Laravel_Test\laravel-sitemap\vendor\meilisearch\meilisearch-php\src\Endpoints\Indexes.php:59
     55▕     {
     56▕         return new self(
     57▕             $this->http,
     58▕             $attributes['uid'],
  ➜  59▕             $attributes['primaryKey'],
     60▕             $attributes['createdAt'],
     61▕             $attributes['updatedAt'],
     62▕         );
     63▕     }

  1   C:\xampp\htdocs\Laravel_Test\laravel-sitemap\vendor\meilisearch\meilisearch-php\src\Endpoints\Indexes.php:59
      Illuminate\Foundation\Bootstrap\HandleExceptions::handleError("Undefined array key "primaryKey"", "C:\xampp\htdocs\Laravel_Test\
laravel-sitemap\vendor\meilisearch\meilisearch-php\src\Endpoints\Indexes.php")

  2   C:\xampp\htdocs\Laravel_Test\laravel-sitemap\vendor\meilisearch\meilisearch-php\src\Endpoints\Indexes.php:89
      MeiliSearch\Endpoints\Indexes::newInstance(["hilinkz-1zcuavQUWzb0mAx7", "enqueued", "indexCreation", "2021-12-11T14:54:06.828356
800Z"])

**Comment by Amik-TJ** [comment]
Using meilisearch version 0.22.0 fixes this problem. If you use the most recent version, you will encounter this problem. However, if you download meilisearch version 0.22.0, it will run smoothly.

**Comment by Kazuto** [comment]
Almost a year and this problem still exists today. You can't install old versions of Meilisearch via Brew (anymore), unfortunately.
